File: Problem/By section/Math/Problem 504/problem_504.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    string convertToBase7(int num) {
        if(!num) return "0";
        
        bool negative = (num < 0);
        string result = "";
        num = negative ? -num : num;
        while(num > 0){
            result = to_string(num % 7) + result;
            num /= 7;
        }
        
        result = negative ? "-" + result : result;
        return result;
    }
};

int main(int argc, char** argv) {
    Solution s;
    string test = "";
    test += '1';
    return 0;
}

File: Problem/By section/Math/Problem 504/problem_504.java
class Solution {
    // Time: O(logn), Space: O(1)
    public String convertToBase7(int num) {
        if(num == 0) return "0";

        boolean is_negative = (num < 0);
        num = is_negative ? -num : num;
        StringBuilder builder = new StringBuilder();
        
        while(num > 0){
            builder.append(num % 7);
            num = (num / 7);
        }

        String ans = builder.reverse().toString();

        return is_negative ? "-" + ans : ans;
    }
}

public class problem_504{
    public static void main(String[] args) {
        
    }
}